tools/cli/domainUtils.go

```diff
@@ -298,6 +298,7 @@ func initializeMetadataMgr(
 	}
 	pFactory := client.NewFactory(
 		&pConfig,
+		nil, // TODO propagate abstract datastore factory from the CLI.
```

Comment:
How do you plan to use CLI to propagate the factory? 
I thought the factory implementation will be needed before/during cadence cluster setup, but no CLI command will work before Cadence are setup successfully.

Maybe what we need is a global CustomDataStoreFactory, and user will SetCustomDataStoreFactory when they start Cadence.

Reply:
CLI would have to be invoked through internal binary, which would populate custom datastore factory, similarly to how main binary works.